vispy/visuals/tube.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import division
+
+from .mesh import MeshVisual
+import numpy as np
+from numpy.linalg import norm
+from ..util.transforms import rotation_matrix
+
+
+class TubeVisual(MeshVisual):
+    """Displays a tube by around a piecewise-linear path.
+
+    The tube mesh is corrected following its Frenet curvature and
+    torsion such that it varies smoothly along the curve, including if
+    the tube is closed.
+
+    Parameters
+    ----------
+    points : ndarray
+        An array of (x, y, z) points describing the path along which the
+        tube will be extruded.
+    radius : float
+        The radius of the tube. Defaults to 1.0.
+    tube_points : int
+        The number of points in the approximate circle of the tube's
+        cross section. Defaults to 8.
+    colors : ndarray | None
```

Comment:
Can you check what we do with other visuals? I think we usually make it so that `colors` can be `ColorArray | Color`, and the visual then deals with it appropriately (an array of colors or a single color).